tests/include/test/helpers.h

```diff
@@ -260,4 +260,14 @@ void mbedtls_test_param_failed_reset_state( void );
 #include "test/fake_external_rng_for_test.h"
 #endif
 
+#if defined(MBEDTLS_TEST_HOOKS)
+/**
+ * \brief   Check that a pure high-level error code is being combined with a
```

Comment:
Thinking about this some more, I am sure that neither high or low level error codes can be positive (greater than zero) however the case of zero is somewhat harder.

Currently low level error codes are allowed to be zero as there are combinations (in `rsa.c` and possibly elsewhere) where a high level error needs to be returned as well as a `ret` which *may or may not* contain a low level error code. I think this makes sense and should be allowed as it significantly reduces the effort to combine error codes in the codebase.

I am not so sure on whether high level error codes should be allowed to be equal to zero however. I cannot see anywhere that it provides benefit as most often a low level function (that can return a low level error) is called by a high level one that will always want to return its own error or will just return the low level error without wanting to do any combination. As this is not specified explicitly by `error.h` though I am thinking it may be easier to make one extra case where `high` is allowed to be zero.

@gilles-peskine-arm what are your thoughts on this?

Reply:
Actually don't worry, currently it is possible to combine zero with a low level error code by doing `0 + MBEDTLS_ERR_XXX` and there is no reason why when combining error codes we cannot do the same even if it isn't currently done in the library. I will add another check for `high == 0` now.

Reply:
Done.